pkg/util/metrics.go

```diff
@@ -0,0 +1,8 @@
+package util
+
+import "github.com/prometheus/client_golang/prometheus"
+
+func RegisterMetric[P prometheus.Collector](reg *prometheus.Registry, collector P) P {
+	reg.MustRegister(collector)
+	return collector
+}
```

Comment:
I think this should probably be merged with `pkg/util/promserver.go`. I like the name `metrics.go` better for the two combined.